File: 01_projects/code/java-playwright/cbaps-dawms/robust/src/main/java/com/playwright/dawms/pages/MilestoneStatusPage.java
package com.playwright.dawms.pages;
import com.microsoft.playwright.*;
import com.playwright.cbaps.library.EnhancedPlaywrightManager;

public class MilestoneStatusPage {
    private Page page;
    private EnhancedPlaywrightManager pwm;
    private Locator milestoneLabel, statusLabel;
    
    public MilestoneStatusPage(Page page, EnhancedPlaywrightManager pwm) {
        this.page = page; this.pwm = pwm;
        this.milestoneLabel = page.locator("#milestone");
        this.statusLabel = page.locator("#status");
        pwm.waitUntilElementVisible("#status");
    }
    
    public String getMilestone() { return pwm.getText(milestoneLabel); }
    public String getStatus() { return pwm.getText(statusLabel); }
    public boolean validateStatus(String expected) { return getStatus().equals(expected); }
    public boolean validateMilestone(String expected) { return getMilestone().equals(expected); }
}

File: 01_projects/code/python-playwright/cbaps-dawms/robust/dawms/pages/milestone_status_page.py
from playwright.sync_api import Page
from loguru import logger

class MilestoneStatusPage:
    def __init__(self, page: Page, pw):
        self.page = page
        self.pw = pw
        self.milestone_label = page.locator("#milestone")
        self.status_badge = page.locator("#status")
        self.pw.wait_for_selector("#status")
    
    def get_milestone(self) -> str:
        return self.pw.get_text(self.milestone_label)
    
    def get_status(self) -> str:
        return self.pw.get_text(self.status_badge)
    
    def validate_status(self, expected: str) -> bool:
        actual = self.get_status()
        valid = actual == expected
        if valid:
            logger.info(f"✅ Status: {expected}")
        return valid
    
    def validate_milestone(self, expected: str) -> bool:
        actual = self.get_milestone()
        return actual == expected
